lib/wasix/src/syscalls/mod.rs

```diff
@@ -1481,14 +1484,49 @@ where
 }
 
 // Function to prepare the WASI environment
-pub(crate) fn _prepare_wasi(wasi_env: &mut WasiEnv, args: Option<Vec<String>>) {
+pub(crate) fn _prepare_wasi(
+    wasi_env: &mut WasiEnv,
+    args: Option<Vec<String>>,
+    envs: Option<Vec<(String, String)>>,
+) {
     // Swap out the arguments with the new ones
     if let Some(args) = args {
         let mut wasi_state = wasi_env.state.fork();
         wasi_state.args = args;
         wasi_env.state = Arc::new(wasi_state);
     }
 
+    // Append the new env vars to the old ones
+    if let Some(envs) = envs {
+        // append new env vars and replace old ones if they exist
+        let mut guard = wasi_env.state.envs.lock().unwrap();
+
+        let mut env_map = guard
+            .iter()
+            .map(|b| {
+                let string = String::from_utf8_lossy(b);
+                let (key, val) = string.split_once('=').expect("env var is malformed");
+
+                (key.to_string(), val.to_string())
+            })
+            .collect::<HashMap<_, _>>();
```

Comment:
Using a HashMap will not preserve the order of ENVs which might cause some bugs. Please use other mechanism instead